checks.d/splunk.py

```diff
@@ -0,0 +1,143 @@
+            self.gauge('splunk.peers.fixup_set', len(peer['content']['fixup_set']), tags=tags)
+            self.gauge('splunk.peers.pending_job_count', peer['content']['pending_job_count'], tags=tags)
+            self.gauge('splunk.peers.primary_count', peer['content']['primary_count'], tags=tags)
+            self.gauge('splunk.peers.primary_count_remote', peer['content']['primary_count_remote'], tags=tags)
+            self.gauge('splunk.peers.replication_count', peer['content']['replication_count'], tags=tags)
+
+            if peer['content']['bucket_count_by_index'] is not None:
+                for bucket, count in peer['content']['bucket_count_by_index'].items():
+                    self.gauge('splunk.peers.bucket_count', count, tags=tags + ['index:{0}'.format(bucket)])
+
+            if peer['content']['status_counter'] is not None:
+                for status, count in peer['content']['status_counter'].items():
+                    self.gauge('splunk.peers.bucket_status', count, tags=tags + ['bucket_status:{0}'.format(status)])
+
+            if status is not None:
+                self.service_check(self.PEER_HEALTH_CHECK_NAME, AgentCheck.CRITICAL,
+                    message=status,
+                    tags = [ 'peer_name:' + host ])
+
+
+    def do_index_metrics(self, url, username, password, timeout):
+        response = self.get_json(url, '/services/cluster/master/indexes', username, password, timeout)
+        for index in response['entry']:
+            name = index['name']
+            searchable = 'true' if index['content']['is_searchable'] == '1' else 'false'
+            # We'll use this as a sigil for index health at the end of the loop
+            status = None
+
+            if searchable == 'false':
+                status = 'Index %s is not searchable.' % (name)
+
+            tags = [
+                'index_name:{0}'.format(name),
+                'searchable:{0}'.format(searchable)
+            ]
+
+            self.gauge('splunk.indexes.size_bytes', index['content']['index_size'], tags=tags)
+            self.gauge('splunk.indexes.total_excess_bucket_copies', index['content']['total_excess_bucket_copies'], tags=tags)
+            self.gauge('splunk.indexes.total_excess_searchable_copies', index['content']['total_excess_searchable_copies'], tags=tags)
+
+            for index_index, index_copy in index['content']['replicated_copies_tracker'].items():
+                index_tags = tags + [ 'copy_index:{0}'.format(index_index) ]
+                actual_copies = int(index_copy['actual_copies_per_slot'])
+                expected_copies = int(index_copy['expected_total_per_slot'])
+                self.gauge('splunk.indexes.replication.actual_copies', actual_copies, tags=index_tags)
+                self.gauge('splunk.indexes.replication.expected_copies', expected_copies, tags=index_tags)
+
+                if actual_copies != expected_copies:
+                    status = 'Index %s is not correctly replicated.'
```

Comment:
I guess it depends on what the action to fix is. As I learned this morning, something like this might be a persistent "Needs more peers in this site" thing, so the fix is "add more peers", which should fix other problems that have arisen for the same reason. I'm not sure of other failure modes. I think a list of indexes might be worth using anyway, because if we're going to present the information I feel like we should present as close to 'correct' information as possible